api/v1alpha1/tenantcontrolplane_types.go

```diff
@@ -249,12 +249,20 @@ type AddonsSpec struct {
 }
 
 // TenantControlPlaneSpec defines the desired state of TenantControlPlane.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.dataStore) || has(self.dataStore)", message="unsetting the dataStore is not supported"
```

Comment:
A CEL policy here would create a breaking change since it requires Kubernetes >= 1.25, and Kamaji has a loose requirement of +1.22 (https://kamaji.clastix.io/reference/versioning/)

@bsctl WDYT here?

Reply:
About version, frankly I don’t care. 1.22 is a very old release. If CEL usage provides benefits over webhook and if we want push in this direction for the future, I think it is a good idea. I would avoid a mix and match of different techniques unless it provides clear benefits, just for sake of maintenance.

Reply:
From my experience, it's convenient to handle as many validation/defaulting tasks as possible using the kubebuilder annotations, i.e. letting the K8s API Server handle that for you. Except for the things that need a more sophisticated logic, then resort to webhooks